include/osmium/thread/pool.hpp

```diff
@@ -162,7 +162,11 @@ namespace osmium {
 
             static constexpr int default_num_threads = 0;
 
-            static Pool& instance() {
+            explicit Pool(int num_threads = default_num_threads) :
```

Comment:
It buys simpler interface. unless it is desired to let user to specify the work queue too, what I have just learned.

(note, in this PR my concern is *number of threads*, other aspects of pool may deserve separate PR, who knows)